extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesDeployer.java

```diff
@@ -65,23 +67,72 @@ public void deploy(KubernetesClientBuildItem kubernetesClient,
+     * When multiple options exist, we choose openshift if it exists, then minikube if it exists.
+     * If none of the above conditions match, we select the single option if there is only one, otherwise we thrown
+     * an {@code IllegalStateException} if there are multiple options to choose from, as there is no way clear way to
+     * pick one. This is better than picking one at random, which because the order in the list isn't fixed would lead
+     * to non-deterministic behavior.
+     */
+    private KubernetesDeploymentTargetBuildItem determineDeploymentTarget(ContainerImageInfoBuildItem containerImageInfo,
+            List<KubernetesDeploymentTargetBuildItem> kubernetesDeploymentTargets, String activeContainerImageCapability) {
+        final KubernetesDeploymentTargetBuildItem selectedKubernetesDeploymentTargetBuildItem;
+        if (kubernetesDeploymentTargets.isEmpty()) {
+            log.debug("No Kubernetes Deployment target was explicitly set. Defaulting to 'kubernetes'.");
+            return new KubernetesDeploymentTargetBuildItem(KUBERNETES, DEPLOYMENT);
+        }
+
+        Optional<KubernetesDeploymentTargetBuildItem> openshiftDeploymentTarget = getOptionalDeploymentTarget(
+                kubernetesDeploymentTargets, OPENSHIFT);
+        boolean checkForMissingRegistry = true;
+        if (openshiftDeploymentTarget.isPresent()) {
+            if (kubernetesDeploymentTargets.size() > 1) {
+                log.info("Multiple Kubernetes deployment targets were set. 'openshift' will be selected.");
+            }
+            selectedKubernetesDeploymentTargetBuildItem = openshiftDeploymentTarget.get();
+            // If we are using s2i there is no need warn about the missing registry
+            checkForMissingRegistry = Capabilities.CONTAINER_IMAGE_S2I.equals(activeContainerImageCapability);
+        } else {
+            Optional<KubernetesDeploymentTargetBuildItem> minikubeDeploymentTarget = getOptionalDeploymentTarget(
+                    kubernetesDeploymentTargets, MINIKUBE);
+            if (minikubeDeploymentTarget.isPresent()) {
+                if (kubernetesDeploymentTargets.size() > 1) {
+                    log.info("Multiple Kubernetes deployment targets were set. 'minikube' will be selected.");
+                }
+                selectedKubernetesDeploymentTargetBuildItem = minikubeDeploymentTarget.get();
+                // No need to check for a registry when using Minikube since Minikube can work with the local docker daemon
+                // Futhermore, the 'ImagePullPolicy' has already been set to 'IfNotPresent' for Minikube
+                checkForMissingRegistry = false;
+            } else if (kubernetesDeploymentTargets.size() > 1) {
+                throw new IllegalStateException(
+                        "Multiple Kubernetes deployment targets present with no known priorities. Please select a single deployment target using the \"quarkus.kubernetes.deploymentTarget\" property.");
```

Comment:
Is it `deployment-target` or `deploymentTarget`?

Reply:
It's obviously the latter :P